pkg/api/core/cloudprovider/aws/certificate_aws.go

```diff
@@ -14,7 +14,7 @@ type certificate struct {
 	provider v1alpha1.CloudProvider
 }
 
-func (c *certificate) DeleteCertificate(opts api.DeleteCertificateOpts) error {
+func (c *certificate) DeleteCognitoCertificate(opts api.DeleteCognitoCertificateOpts) error {
```

Comment:
Why does Cognito require its own unique certificate deletion functionality?

Reply:
I am not sure, this functionality existed prior to my refactor, I just moved it. My assumption was that the auth cert had to be disassociated before cognito could be deleted